src/PrestaShopBundle/Service/Routing/Router.php

```diff
@@ -96,6 +96,12 @@ public static function generateTokenizedUrl($url, $token)
 
         $queryParams['_token'] = $token;
 
-        return $baseUrl . '?' . http_build_query($queryParams, '', '&');
+        $url = $baseUrl . '?' . http_build_query($queryParams, '', '&');
+        if (isset($components['fragment']) && $components['fragment'] !== '') {
```

Comment:
Ok so I admit if Symfony uses the term `fragment` I have nothing to argue anymore 😅

This `generateTokenizedUrl` is quite complicated on a side note, I wonder why we need this, it seems like we could simply add a `_token` parameter in the array used in our `generate` method and let the Symfony router handle it all But that's not the point of this PR anyway

Reply:
You won't need it once all PrestaShop will be migrated, for now you need a system compatible with Symfony and the legacy Security system => https://github.com/PrestaShop/PrestaShop/pull/5912

Reply:
Hi! Thanks for the tip, however, we think we may have to remove this system as we think it may have flaws.
A CSRF token has to change on each request, shouldn't be predictable and of course, shouldn't be reused between requests.
Unfortunately, the legacy implementation has lack of security and is not secure following the above criterion:
https://github.com/PrestaShop/PrestaShop/pull/5912/files#diff-979b13751a5046d872b3d058133775d5563c5e9d0571fc9adf4a588a4dc6517bR48

It's based on the Username, not on the page, or the request, or the time, which means, you could easily forge a request by getting it in the URL, then create an ajax request, and get access.
So we'll need something stronger